routes/index.js

```diff
@@ -1,367 +1,376 @@
+  // and provide it to the view to display
+  const profile = {}
+ 	return res.render('account.hbs', profile)
+}
+
+exports.save_account_details = function(req, res, next) {
+  // get the profile details from the JSON
+	const profile = req.body
+  // validate the input
+  if (validator.isEmail(profile.email, { allow_display_name: true })
+    // allow_display_name allows us to receive input as:
+    // Display Name <email-address>
+    // which we consider valid too
+    && validator.isMobilePhone(profile.phone, 'he-IL')
+    && validator.isAscii(profile.firstname)
+    && validator.isAscii(profile.lastname)
+    && validator.isAscii(profile.country)
+  ) {
+    // trim any extra spaces on the right of the name
+    profile.firstname = validator.rtrim(profile.firstname)
+    profile.lastname = validator.rtrim(profile.lastname)
+
+    // render the view
+    return res.render('account.hbs', profile)
+  } else {
+    // if input validation fails, we just render the view as is
+    console.log('error in form details')
+    return res.render('account.hbs')
+  }
+}
+
+exports.isLoggedIn = function (req, res, next) {
+  if (req.session.loggedIn === 1) {
+    return next()
+  } else {
+    return res.redirect('/')
+  }
+}
+
+exports.logout = function (req, res, next) {
+  req.session.loggedIn = 0
+  req.session.destroy(function() { 
+    return res.redirect('/')  
+  })
+}
+
+function parse(todo) {
+  var t = todo;
+
+  var remindToken = ' in ';
+  var reminder = t.toString().indexOf(remindToken);
+  if (reminder > 0) {
+    var time = t.slice(reminder + remindToken.length);
+    time = time.replace(/\n$/, '');
+
+    var period = hms(time);
+
+    console.log('period: ' + period);
+
+    // remove it
+    t = t.slice(0, reminder);
+    if (typeof period != 'undefined') {
+      t += ' [' + ms(period) + ']';
+    }
+  }
+  return t;
+}
+
+exports.create = function (req, res, next) {
+  // console.log('req.body: ' + JSON.stringify(req.body));
+
+  var item = req.body.content;
+  var imgRegex = /\!\[alt text\]\((http.*)\s\".*/;
+  if (typeof (item) == 'string' && item.match(imgRegex)) {
+    var url = item.match(imgRegex)[1];
+    console.log('found img: ' + url);
+
+    execFile('identify', [url], function (err, stdout, stderr) {
+      console.log(err);
+      if (err !== null) {
+        console.log('Error (' + err + '):' + stderr);
+      }
+    });
+
+  } else {
+    item = parse(item);
+  }
+
+  new Todo({
+    content: item,
+    updated_at: Date.now(),
+  }).save(function (err, todo, count) {
+    if (err) return next(err);
+
+    /*
+    res.setHeader('Data', todo.content.toString('base64'));
+    res.redirect('/');
+    */
+
+    res.setHeader('Location', '/');
+    res.status(302).send(todo.content.toString('base64'));
+
+    // res.redirect('/#' + todo.content.toString('base64'));
+  });
+};
+
+exports.destroy = function (req, res, next) {
+  Todo.findById(req.params.id, function (err, todo) {
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg" width="20"/></sub>&nbsp; NoSQL Injection

Unsanitized input from an HTTP parameter flows into findById, where it is used in an NoSQL query. This may result in an NoSQL Injection vulnerability.

<h6>
<a href="https://cwe.mitre.org/data/definitions/943.html">CWE-943</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 860</a> | Line 200</h6>